### DISCUSSION TITLE
我發現Python Quiz上的答案錯誤

### DISCUSSION THREAD

**Discussion by TobywithDino** [1101-GS4719-C/Python-Community-GS4719-C#1587] [General]
在9/20的課程中，我在練習Python Quiz上面的題目
發現[數字變數第30題](https://ncuedu.tw/learning/chatgpt/cid/1/qid/30)
四個選項中，沒有一個是真正的答案
當我把題目中的程式直接拿去跑
結果會是：3.0 3.5 1.5
而第30題給的答案是C：3 3.5 1.5
我才發現，雖然運算子 // 是整除，例如 5//2 的結果會是 2 而不是2.0
但當運算子左右數字的型態為浮點數時(有小數點)，其結果雖然也會是整除後的數字，但型態會自動轉換成浮點數(n.0)
當天發現這件事的我直接跟老師反應了，現在網站上的答案已被修正。
同樣的問題在[數字變數第32題](https://ncuedu.tw/learning/chatgpt/cid/1/qid/32)也有出現，答案應為7.0，AI似乎不完全理解Python中//的小規則。

**Comment by FakeWood** [comment]
哇，謝謝同學指出錯誤，AI生成的題目難免有點小錯誤，還是需要大家的做題的時候多多留意呢!
相信在大家的回報下，這個題庫也會愈來愈完善!
突然發現我這樣寫好像老師喔，但我不是哈哈
以後在上面做題目的時候，我也會多多留心有沒有什麼錯誤的地方!

**Comment by 1chooo** [comment]
若你希望 `5 // 2` 會得到 `2`，若想要得到 `2.0` 的答案，可以改成 `5 // 2.0` 或是 `5.0 // 2` 甚至是 `5.0 // 2.0` 都會得到哦！

```py
print(5 // 2)       # 2
print(5.0 // 2)     # 2.0
print(5 // 2.0)     # 2.0
print(5.0 // 2.0)   # 2.0
```

By TA

以下 By ChatGPT
1. 如果兩個操作數都是整數，則結果將是整數，且返回的是除法結果的最大整數部分。小數部分將被捨棄。
2. 如果至少有一個操作數是浮點數，則結果將是浮點數，且返回的是除法結果的最大整數部分。小數部分將被捨棄。

因此，`//` 運算符的行為是根據操作數的型別來決定的。如果所有操作數都是整數，則結果也將是整數。如果有至少一個操作數是浮點數，則結果將是浮點數。
